### PR TITLE
Longest Palindrome Substring

### PR DESCRIPTION
**Pull request by achalesh27022003**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1999

### CODE REVIEW

File: C-Plus-Plus/dp/LongestPalindromeString.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+In our approach, we use a 2D Boolean array to store the results.
```
Comment (atarax665) [inline C-Plus-Plus/dp/LongestPalindromeString.cpp:2]: Add a beginner pitch of the problem too.

File: C-Plus-Plus/dp/LongestPalindromeString.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+*/
+
+#include<iostream>
+#include<string>
+using namespace std;
+
+string DynamicProgrammingGetLongestPalindrome(string input_str)
```
Comment (atarax665) [inline C-Plus-Plus/dp/LongestPalindromeString.cpp:9]: The function name shouldn't be this long.

File: C-Plus-Plus/dp/LongestPalindromeString.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    cout<<"The longest palindromic substring is " << resultstring <<endl;
+}
+
+/*
+Sample Input : Enter a String
+               ptrabbart
+Sample Output : The longest palindromic substring is trabbbart
+Time complexity : O(n^2)
+Space complexity : O(n^2)
+*/
```
Comment (atarax665) [inline C-Plus-Plus/dp/LongestPalindromeString.cpp:58]: Add a line at the EOF.

File: C-Plus-Plus/dp/LongestPalindromeString.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    if(j-i+1>palindromeLength)
+                    {
+                       palindromeLength = j-i+1; 
+                       resultstring = input_str.substr(i, j+1);
+                    }
+            }
+        }
+    }
+    return resultstring;
+
```
Comment (atarax665) [inline C-Plus-Plus/dp/LongestPalindromeString.cpp:38]: Remove all these unnecessary blank lines.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Do not remove any existing files. Please revert those changes.

File: C-Plus-Plus/dp/LongestPalindromeString.cpp (hunk 1 above)
Comment (achalesh27022003) [reply C-Plus-Plus/dp/LongestPalindromeString.cpp:2]: done

**Comment by atarax665** [comment]
No update!

**Pull request closed without merging**